src/libsyntax/diagnostic.rs

```diff
@@ -293,8 +293,8 @@ impl fmt::Display for Level {
 
         match *self {
             Bug => "error: internal compiler error".fmt(f),
-            Fatal | Error => "error".fmt(f),
-            Warning => "warning".fmt(f),
+            Fatal | Error => "\n  error".fmt(f),
+            Warning => "\n  warning".fmt(f),
```

Comment:
I've put this here only for a preview. If we're gonna avoid the file path for the other messages too, then we can put it directly into the function.